### ISSUE TITLE
'e18e/prefer-array-fill' incorrectly errors when the callback is not constant

### ISSUE THREAD

**Issue by yharaskrik** [e18e/eslint-plugin#71]
We use Array.fill for example this way:

```
const veryLongItems = Array.from({ length: 5 }, () => faker.lorem.sentences(3));
```

This ensures that the array elements are all different sentences. Right now the plugin is determining that that breaks the rule even though it is not a constant value.

For example this IS wrong:

```
const sentence =  faker.lorem.sentences(3);

const veryLongItems = Array.from({ length: 5 }, () => sentence);
```

Since the `sentence` never changes, but when using a callback with a function call inside that _could_ return a different value, that should not break the rule.

**Comment by yharaskrik** [comment]
> i wonder if it is too unsafe to move call expressions and we should just exclude them 🤔
> 
> what do you think?
> 
> if we just exclude call expressions from the rule, it'd only catch constants/literals basically (like `() => foo`)

Hey, not exactly sure what you mean.

We may get some false negatives if we do not flag factory functions with callables but we get false positives if we do, not sure what is prefeerable.
